lib/msal-common/src/uCache/serialize/Deserializer.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AccessTokenEntity } from "../entities/AccessTokenEntity";
+import { IdTokenEntity } from "../entities/IdTokenEntity";
+import { RefreshTokenEntity } from "../entities/RefreshTokenEntity";
+import { AccountEntity } from "../entities/AccountEntity";
+import { AppMetadataEntity } from "../entities/AppMetadataEntity";
+import { CacheHelper } from "../utils/CacheHelper";
+import {
+    AccessTokenCacheMaps,
+    IdTokenCacheMaps,
+    RefreshTokenCacheMaps,
+    AccountCacheMaps,
+    AppMetadataCacheMaps
+} from "../serialize/JsonKeys";
+import { AccessTokenCache, IdTokenCache, RefreshTokenCache, AccountCache, AppMetadataCache, StringDict } from "../../utils/MsalTypes";
+
+// TODO: Can we write this with Generics?
+export class Deserializer {
+    /**
+     * Deserializes access tokens to AccessTokenEntity objects
+     * @param accessTokens
+     */
+    static deSerializeAccessTokens(accessTokens: StringDict): AccessTokenCache {
+        const atObjects = {};
+        Object.keys(accessTokens).map(function(key) {
+            const mappedAT = CacheHelper.renameKeys(
+                accessTokens[key],
+                AccessTokenCacheMaps.fromCacheMap
+            );
+            const accessToken: AccessTokenEntity = new AccessTokenEntity();
+            accessToken.toObject(mappedAT);
+            atObjects[key] = accessToken;
+        });
+
+        return atObjects;
```

Comment:
This can also be done with `reduce`:

```js
return Object.keys(accessTokens).reduce((accessTokens, key) => {
   const mappedAT = CacheHelper.renameKeys(accessTokens[key], AccessTokenCacheMaps.fromCacheMap);
   const accessToken = new AccessTokenEntity();
   accessToken.toObject(mappedAT);
   return {
      ...accessTokens,
      [key]: accessToken
   };
}, {});

Reply:
The methods below also can use `reduce`.

Reply:
I tried reduce and ran into some issues. Can check offline with you and change it. Is there a preference between map and reduce in general?

Reply:
Sticking to map.